provider/win_driver_installer_test.py

```diff
@@ -525,3 +526,40 @@ def viofs_basic_io_test(test, params, vm):
         test.fail('The md5 value of host is not same to guest.')
     else:
         test.log.info("The md5 of host is as same as md5 of guest.")
+
+
+def balloon_test(test, params, vm, balloon_test_win):
+    """
+    Balloon service test in Windows guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param vm: The vm object
+    :param balloon_test_win: Basic functions for memory ballooning testcase
+    """
+    error_context.context("Running balloon service test...", LOG_JOB.info)
+    mem_check = params.get("mem_check", "yes")
+    mem_stat_working = True
+
+    session = vm.wait_for_login()
+    error_context.context("Config balloon service in guest", test.log.info)
+    balloon_test_win.configure_balloon_service(session)
+    min_sz, max_sz = balloon_test_win.get_memory_boundary()
+
+    tag = "envict"
+    error_context.context("Running %s test" % tag, test.log.info)
+    expect_mem = int(random.uniform(min_sz,
+                                    balloon_test_win.get_ballooned_memory()))
+    balloon_test_win.run_ballooning_test(expect_mem, tag)
+    time.sleep(20)
```

Comment:
Could we sleep time dynamically? Because it will waste lots of time if we wanna repeat such a test.